### PR TITLE
W-11287187 adding CLI error docs

### PR DESCRIPTION
**Pull request by IsaacEldridge**

### CODE REVIEW

**Review by Extor3D — APPROVED** [review]
Everything looks great!

File: modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -34,4 +34,5 @@
 * xref:configure-apq-cdn.adoc[Using Persisted Queries and CDNs]
 * xref:overview-cli-extensions.adoc[Using the DataGraph CLI and API Extensions]
 ** xref:api-extensions.adoc[API Extensions for DataGraph]
-** xref:datagraph-cli.adoc[DataGraph CLI Reference]
\ No newline at end of file
+** xref:datagraph-cli.adoc[DataGraph CLI Reference]
+** xref:datagraph--cli-errors.adoc[DataGraph CLI Errors]
```
Comment (Cristian-Venticinque) [inline modules/ROOT/nav.adoc:38]:
```suggestion
** xref:datagraph-cli-errors.adoc[DataGraph CLI Errors]
```

File: modules/ROOT/pages/datagraph-cli-errors.adoc [hunk 2]
```diff
@@ -0,0 +1,227 @@
+= DataGraph CLI Errors
+
+When using the Datagraph CLI, you might encounter errors. The errors are divided in two categories:
+
+. Command errors: These are errors you can encounter when using incorrect arguments or options in commands.
+. Validation errors: These are errors you can encounter with validation when adding new API sources or when updating sources in the unified schema.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/datagraph-cli-errors.adoc:6]: These are errors you can encounter with validation when > These errors occur during validation when...

File: modules/ROOT/pages/datagraph-cli-errors.adoc [hunk 3]
```diff
@@ -0,0 +1,227 @@
+= DataGraph CLI Errors
+
+When using the Datagraph CLI, you might encounter errors. The errors are divided in two categories:
+
+. Command errors: These are errors you can encounter when using incorrect arguments or options in commands.
+. Validation errors: These are errors you can encounter with validation when adding new API sources or when updating sources in the unified schema. 
+
+== Command Errors
+
+DataGraph returns these errors when you don't correctly define a command. These errors can be caused by an incorrect option
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/datagraph-cli-errors.adoc:10]: don't correctly > incorrectly

File: modules/ROOT/pages/datagraph-cli-errors.adoc [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  }
+}
+----
+
+Cause:
+
+This error indicates that at least one option is required to run the command.
+
+* `datagraph sources add`: Either the `--url` option or the `--discover-url` option is required for this command.
+* `datagraph sources update/promote`: Any available option is required. You must define an element to update or promote in the target source.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/datagraph-cli-errors.adoc:29]: You must define > Define

File: modules/ROOT/pages/datagraph-cli-errors.adoc [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    "errorMessage": "Must define url or set --discover-url"
+  }
+}
+----
+
+Cause:
+
+This error indicates that you must set the URL using either `--url` or `--discover-url`.
+
+* `datagraph sources add`: Since an URL is a required field in the `add` command, you must set one of the two options, `--url` or `--discover-url`.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/datagraph-cli-errors.adoc:114]: Since > Because

File: modules/ROOT/pages/datagraph-cli-errors.adoc [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  "errorCode": "dg-federation-source-not-in-env",
+  "errorMessage": "API not found in current environment"
+}
+----
+
+Cause:
+
+This error indicates that the API source you are trying to edit, delete, or update exists, but it is not in the working environment.
+
+* `datagraph sources update/promote`: The UUID sent is not in the working environment. Or if you are using the `--version` option, the major version is not compatible with the current version.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/datagraph-cli-errors.adoc:162]:
```suggestion
* `datagraph sources update/promote`: The UUID sent is not in the working environment. Or, if you are using the `--version` option, the major version is not compatible with the current version.
```

File: modules/ROOT/pages/datagraph-cli-errors.adoc [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+== Validation Errors
+
+Validation errors occur when you use the `datagraph validate asset` command before adding or updating an API source against the unified schema.
+
+=== Editing Errors
+
+Editing errors are generated when the edits in the particular API source contain a semantic problem. These errors use the code property `dg-federation-customization-conflict` and have the following structure:
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/datagraph-cli-errors.adoc:170]: Editing errors are generated when > DataGraph returns these errors when... (to sync with == Command Errors)

File: modules/ROOT/pages/datagraph-cli-errors.adoc (hunk 3 above)
Comment (IsaacEldridge) [reply modules/ROOT/pages/datagraph-cli-errors.adoc:10]: lordy

**Review by Cristian-Venticinque — APPROVED** [review]

**Pull request merged**